### PR TITLE
Specify capacity for the slice created in call.HeaderNames()

### PR DESCRIPTION
**Pull request by ztstewart**
- `call.HeaderNames()` returns a slice that contains the names of every header in the internal header map. However, it does not specify a capacity; it therefore allocates a bit more than expected. This change simply adds a capacity value, which should speed things up quite a bit.

### CODE REVIEW

**Comment by ztstewart** [comment]
Thanks @kriskowal for the speedy review!

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
-- No changes yet
+### Fixed
+- call.HeaderNames() now creates specifies a capacity when creating a slice,
```
Comment (AlexAPB) [inline CHANGELOG.md]: `now creates specifies` -> `now specifies`

**Comment by AlexAPB** [comment]
Thanks! @ztstewart Can you add a more descriptive PR title?
After the CHANGELOG fix (and build/tests pass), I'll merge this.

**Comment by ztstewart** [comment]
@AlexAPB absolutely. I've updated the title to something that is hopefully more descriptive than the branch name.

Thanks for fixing the typo in the CHANGELOG also

File: CHANGELOG.md (hunk 1 above)
Comment (ztstewart) [reply CHANGELOG.md]: Thanks for finding this; I've updated the diff to fix the typo.

**Comment by AlexAPB** [comment]
Thanks @ztstewart!

**Pull request merged**